virttest/env_process.py

```diff
@@ -722,12 +722,14 @@ def preprocess(test, params, env):
             kvm_userspace_version = "Unknown"
     else:
         qemu_path = utils_misc.get_qemu_binary(params)
-        version_output = avocado_process.system_output("%s -help" % qemu_path,
+        version_output = avocado_process.system_output("%s -version" % qemu_path,
                                                        verbose=False)
         version_line = version_output.split('\n')[0]
-        matches = re.findall("[Vv]ersion .*?,", version_line)
+        regex = r"QEMU emulator version\s([0-9]+\.[0-9]+\.[0-9]+)\s?\((.*)\)"
```

Comment:
We have the different pattern for qemu 0.12.*:
```c
static void version(void)
{
    printf("QEMU PC emulator version " QEMU_VERSION QEMU_PKGVERSION ", Copyright (c) 2003-2008 Fabrice Bellard\n");
}
```

So I'd suggest adding `(?:PC )?` into the regex.

Reply:
also the 2.7 `qemu-system-i386` does not uses the `\s`:

```
QEMU emulator version 2.7.1(qemu-2.7.1-7.fc25), Copyright (c) 2003-2016 Fabrice Bellard and the QEMU Project developers
```

Reply:
@luckyh Yes for adding "PC".  @ldoktor the "\s" is optional in the proposed regex (`\s?`), and your example is (AFAICT) like the second example I tested it with.

Anyway, I'll also add a test for that regex.

Reply:
Oh true, I saw `version\s(` and forgot it's the separator between `version` and `2...`.